remote_config/src/common.h

```diff
@@ -63,6 +66,25 @@ class FutureData {
   static FutureData* s_future_data_;
 };
 
+template <typename TConfigUpdate, typename TRemoteConfigError>
```

Comment:
What is this LambdaConfigUpdateListener class trying to accomplish compared to calling the callback directly? It should be possible to pass around a std::function variable if needed.

Reply:
Just providing a wrapper for the std::function. When I tried to call the std::function in the iOS implementation and I passed in ConfigUpdate and RemoteConfigError values, I got an error when I tried to build the project. The error I got were along the lines of `candidate function not viable: no known conversion from 'long' to 'firebase::remote_config::ConfigUpdate' for 1st argument`

Reply:
I assume that you are referring to LambdaEventListener, but I think that class is used to listen to multiple types of events (QuerySnapshot, DocumentSnapshot, or LoadBundleTaskProgress), so a templated type makes sense.

Here we expect TConfigUpdate to always be ConfigUpdate, so it should be possible to just use the types ConfigUpdate and RemoteConfigError directly.

The error about converting from long to ConfigUpdate is strange. I suspect that there may have been some other issue causing that, like a pointer being provided instead of the object itself, but it is hard to say without seeing the code. Do you recall the lines of code causing that error?

Anyways, I think this class should not be a templated type. It is probably fine to wrap std::function in a class like this, at least for now.

Reply:
I got this error for both parameters RemoteConfigError and ConfigUpdate when I tried to invoke it in the iOS implementation. The code that caused this was: 
`
registration = [impl() addOnConfigUpdateListener: ^(FIRRemoteConfigUpdate *_Nullable update,
                                  NSError *_Nullable error) {
          if (error) {
            config_update_listener(nullptr, ConvertFIRRemoteConfigUpdateError(error));
          }
          if (update) {
            config_update_listener(ConvertConfigUpdateKeys(update.updatedKeys), nullptr);
          }
    }];


`

Reply:
After looking back at it I think you're right and I didn't need the wrapper. Removed it and use the function directly

Reply:
Ok, the error there would be that ConfigUpdate is a struct which is not a nullable type. It looks like you figured that out though with the later code. Calling the function directly looks good.